File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = defaultdict(int)
        for i,n in enumerate(nums):
            diff = target-n
            if diff in prevMap:
                return (prevMap[diff], i)
            
            prevMap[n] = i

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> umap;
        vector<int> res;
        for(int i=0; i<nums.size(); i++)
        {
            int diff = target-nums[i];
            if(umap.find(diff)!=umap.end())
            {
                res.push_back(umap[diff]);
                res.push_back(i);
                return res;
            }
            umap[nums[i]] = i;
                
        }
        return res;
    }
};